### PR TITLE
[zelos] Add right connection shape element.

### PR DESCRIPTION
**Pull request by samelhusseini**
## The basics
- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Also fixes https://github.com/google/blockly/issues/3481 but as a side effect to inheriting finalize_.

### Proposed Changes

So far we've been adding extra spacing for the right connection shape without actually adding an element. Add an element and have it match the dimensions of the output connection.

We might want to consider moving away from just adding custom logic for the output connection and adding a left side shape as well. @rachel-fenichel 

### Reason for Changes

Zelos rendering.

### Test Coverage

Screen Shot 2019-12-05 at 10 56 09 AM

Tested on:

### CODE REVIEW

File: tests/rendering/zelos/zelos.html [hunk 1]
```diff
@@ -33,6 +33,10 @@
     goog.require('Blockly.blockRendering.Debug');
     goog.require('Blockly.zelos.Renderer');
     // Blockly.blockRendering.startDebugger();
+
+    Blockly.BlockSvg.prototype.setOutputShape = function() {
```
Comment (rachel-fenichel) [inline tests/rendering/zelos/zelos.html]: Add a comment (here or in the other PR) that this is for testing with the PXT blocks.

File: core/renderers/zelos/measurables/row_elements.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Objects representing elements in a row of a rendered
```
Comment (rachel-fenichel) [inline core/renderers/zelos/measurables/row_elements.js]: Update fileoverview.

File: core/renderers/zelos/info.js [hunk 3]
```diff
@@ -259,42 +246,51 @@ Blockly.zelos.RenderInfo.prototype.adjustXPosition_ = function() {
...
+ * output connection.
+ * @protected
  */
-Blockly.zelos.RenderInfo.prototype.finalize_ = function() {
-  // Performance note: this could be combined with the draw pass, if the time
-  // that this takes is excessive.  But it shouldn't be, because it only
-  // accesses and sets properties that already exist on the objects.
+Blockly.zelos.RenderInfo.prototype.finalizeOutputConnection_ = function() {
   var yCursor = 0;
+  var firstInpurRow = null;
```
Comment (rachel-fenichel) [inline core/renderers/zelos/info.js]: Spelling: `firstInpurRow` -> `firstInputRow`

File: core/renderers/zelos/info.js [hunk 4]
```diff
@@ -259,42 +246,51 @@ Blockly.zelos.RenderInfo.prototype.adjustXPosition_ = function() {
...
-    this.outputConnection.setShapeDimensions(
-        this.outputConnection.shape.height(this.height),
-        this.outputConnection.shape.width(this.height));
+    var connectionHeight = this.outputConnection.shape.height(yCursor);
+    var connectionWidth = this.outputConnection.shape.width(yCursor);
 
-    // Recompute the bounds as we now know the output connection dimensions.
-    this.computeBounds_();
-  }
+    this.outputConnection.height = connectionHeight;
```
Comment (rachel-fenichel) [inline core/renderers/zelos/info.js]: I agree that the output connection should become a measurable and this should move into it.

File: core/renderers/zelos/info.js [hunk 5]
```diff
@@ -259,42 +246,51 @@ Blockly.zelos.RenderInfo.prototype.adjustXPosition_ = function() {
...
+    this.outputConnection.startX = connectionWidth;
 
-  var widestRowWithConnectedBlocks = 0;
-  for (var i = 0, row; (row = this.rows[i]); i++) {
-    row.xPos = this.startX;
+    var rightConnection =
+        new Blockly.zelos.RightConnectionShape(this.constants_);
+    rightConnection.height = connectionHeight;
+    rightConnection.width = connectionWidth;
+    firstInpurRow.elements.push(rightConnection);
```
Comment (rachel-fenichel) [inline core/renderers/zelos/info.js]: Does this have to go onto the row?  I would prefer to store a rightSide and a leftSide measurable.  Consider multi-row reporter blocks for PXT.

Comment (samelhusseini) [reply core/renderers/zelos/info.js]: Yeah I agree, for now I'll add the measurable on the info object, but I think we could see ourselves creating a Col object similar to Row. and having the left and ride side be a column.

**Comment by samelhusseini** [comment]
Updated with a right side measurable.

File: core/renderers/zelos/info.js [hunk 6]
```diff
@@ -82,6 +83,13 @@ Blockly.zelos.RenderInfo = function(renderer, block) {
    * @override
    */
   this.isInline = true;
+
+  /**
+   * An object with rendering information about the right connection shape.
+   * @type {!Blockly.zelos.RightConnectionShape}
```
Comment (rachel-fenichel) [inline core/renderers/zelos/info.js]: Fix type: this is very definitely nullable.

Comment (samelhusseini) [reply core/renderers/zelos/info.js]: Yep, just fixed travis.

**Pull request merged**